### PR TITLE
4.12.0 patch notes

### PR DESCRIPTION
**Pull request by ScottSuarez**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,22 @@
 ## 4.12.0 (Unreleased)
+NOTE:
+* updated to go 1.16.14 ([#11132](https://github.com/hashicorp/terraform-provider-google/pull/11132))
+
+DEPRECATIONS:
+* datafusion: deprecated `service_account` in `google_datafusion_instance`. Use `tenant_project_id` instead to extract the tenant project ID (beta) ([#11087](https://github.com/hashicorp/terraform-provider-google/pull/11087))
```
Comment (c2thorn) [inline CHANGELOG.md:6]: beta-only field?

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ScottSuarez) [reply CHANGELOG.md]: removed

**Review by c2thorn — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,19 @@
 ## 4.12.0 (Unreleased)
+NOTE:
+* updated to go 1.16.14 ([#11132](https://github.com/hashicorp/terraform-provider-google/pull/11132))
+
+IMPROVEMENTS:
+* bigquery: added support for authorized datasets to `google_bigquery_dataset.access` and `google_bigquery_dataset_access` ([#11091](https://github.com/hashicorp/terraform-provider-google/pull/11091))
+* bigtable: added `multi_cluster_routing_cluster_ids` fields to `google_bigtable_app_profile` ([#11097](https://github.com/hashicorp/terraform-provider-google/pull/11097))
+* compute: updated `instance` attribute for `google_compute_network_endpoint` to be optional, as Hybrid connectivity NEGs use network endpoints with just IP and Port. ([#11147](https://github.com/hashicorp/terraform-provider-google/pull/11147))
+* compute: added `NON_GCP_PRIVATE_IP_PORT` value for `network_endpoint_type` in the `google_compute_network_endpoint_group` resource ([#11147](https://github.com/hashicorp/terraform-provider-google/pull/11147))
+* compute: added `provisioning_model` field to `google_compute_instance_template ` resource to support Spot VM(beta) ([#11049](https://github.com/hashicorp/terraform-provider-google/pull/11049))
```
Comment (shuyama1) [inline CHANGELOG.md:10]: Looks like this is also a beta only feature. Should we also exclude it from this changelog?

Comment (ScottSuarez) [reply CHANGELOG.md]: nice catch !

**Pull request merged**